app/Database/Repositories/Eloquent/UserRepository.php

```diff
@@ -62,6 +62,27 @@ public function all($sortBy = 'created_at', $sortDir = 'asc', $perPage = 10)
     {
         return $this->userModel->orderBy($sortBy, $sortDir)->paginate($perPage);
     }
+    
+    /**
+     * Get all users who match a certain set of criteria.
+     *
+     * @param string username
+     * @param string email
+     * @param int role_id
+     * @param string $sortBy
+     * @param string $sortDir
+     * @param int $perPage
+     *
+     * @return mixed
+     */
+    public function search($username = "", $email = "", int $role_id = 0, $sortBy = 'created_at', $sortDir = 'asc', $perPage = 10)
```

Comment:
Since we're using PHP 7, could you type hint `username`, `email`, etc. like you have for `role_id`? Also, `role_id` should be `roleId` as we use `camelCase` for variable names.